app/gui/enso-profiler-enso-data/src/beanpole.rs

```diff
@@ -0,0 +1,162 @@
+//! A diagram type for illustrating timings of messages between processes.
+
+
+
+// ===============
+// === Diagram ===
+// ===============
+
+/// The data necessary to create a diagram of message timings.
+#[derive(Debug, Default)]
+pub struct Diagram<'a> {
+    processes: Vec<&'a str>,
+    messages:  Vec<Message>,
+}
+
+impl<'a> Diagram<'a> {
+    /// Define a new process. Return a handle.
+    pub fn process<'b: 'a>(&mut self, name: &'b str) -> Process {
+        let id = self.processes.len();
+        self.processes.push(name);
+        Process(id)
+    }
+
+    /// Log a message between two processes.
+    pub fn message(&mut self, p0: Process, p1: Process, time: f64, label: String) {
+        self.messages.push(Message { p0, p1, time, label });
+    }
+}
+
+
+// === Process ===
+
+/// A process that may send and receive messages.
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+pub struct Process(usize);
+
+
+// === Message ===
+
+/// An event of communication between processes.
+#[derive(Clone, Debug)]
+pub struct Message {
+    p0:    Process,
+    p1:    Process,
+    time:  f64,
+    label: String,
+}
+
+
+
+// ====================
+// === Svg rendering ==
+// ====================
+
+/// Rendering a [`Diagram`] in SVG format.
+pub mod svg {
```

Comment:
Couldyou show on screenshot in this PR this diagram? Such things are amazing in PR description - they are very informative and allow readers to really fast get what this PR introduces.

Reply:
There is a link to an example output in the PR description: https://user-images.githubusercontent.com/1047859/163629928-c017cecf-2540-47a8-a481-820e7c1212b1.svg

Reply:
Added a screenshot, because github doesn't show SVGs inline.